File: Baekjoon/Class2/34.11866 요세푸스 문제 0.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
people = list(range(1, n + 1))
out = []

idx = 0
while people:
    idx = (idx + k - 1) % len(people) #k - 1번 이동
    out.append(str(people.pop(idx))) #k번째 사람
print(f"<{", ".join(out)}>")

''' deque rotate 사용
from collections import deque
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
people = deque(range(1, n + 1))
out = []

while people:
    people.rotate(-(k - 1)) #k - 1번 rotate left
    out.append(people.popleft()) #k번째 사람
print(f"<{", ".join(out)}>")
'''

File: Baekjoon/Class2/34.11866 요세푸스 문제 0.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split(" ").map(Number);

const [n, k] = input;
const people = Array.from({ length: n }, (_, i) => i + 1);
const out = [];

let idx = 0;
while(people.length !== 0) {
    idx = (idx + k - 1) % people.length; // k - 1번 이동
    out.push(people.splice(idx, 1)[0]); // k번째 사람
}
console.log(`<${out.join(", ")}>`);